app/models/task.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class Task < ApplicationRecord
+  belongs_to :case_record
+  belongs_to :parent, polymorphic: true
+
+  has_many :tasks, as: :parent, class_name: "Task", dependent: :destroy, autosave: true
+
+  validates :slug, presence: true, uniqueness: {scope: :case_record_id}
+  validates :name, presence: true, uniqueness: {scope: [:parent_type, :parent_id], message: "must be unique within the same group"}
```

Comment:
I'd avoid the uniqueness validation and just rely on the unique index to blow up - these tasks aren't being created in a form so there's no need for user feedback error messages and it avoids extra queries.